### ISSUE TITLE
Make options more distribution neutral

### ISSUE THREAD

**Issue by thkukuk** [coreos/ignition#646]
# Feature Request #
Options/names like coreos.firstboot are fine if you think about a whole distribution, but if it is about making ignition useable for a wide range of distributions, it will be quite confusing.

It would be nice to either change the names it "ignition" or use something more generic.

## Environment ##
openSUSE Tumbleweed, e.g.

## Desired Feature ##

It would be nice to either change the names it "ignition" or use something more generic.

**Comment by dustymabe** [comment]
Migrated to https://github.com/dustymabe/ignition-dracut/issues/24 since that is where we would need to make the change for `coreos.firstboot` language.

**Comment by bgilbert** [comment]
`coreos.firstboot` is not part of Ignition, but Ignition hardcodes `coreos` in other places:

- The `coreos.config.url` kernel option,
- The VMware Guestinfo variables `coreos.config.data` and `coreos.config.data.encoding`,
- The content type `application/vnd.coreos.ignition+json` (probably fine?).

**Comment by ajeddeloh** [comment]
We can't drop support for the old ones (for compatibility) but we can add aliases and support both the historic `coreos.*` and a new `ignition.*`. Thoughts?

**Comment by bgilbert** [comment]
@ajeddeloh If we make the changes for spec 3.0, which will run on a disjoint set of operating systems, what compatibility constraints do you expect we'll still need to respect?

**Comment by ajeddeloh** [comment]
We can absolutely drop support for 3.0.0 spec. But more immediately we can add aliases and deprecate the old names.

**Comment by thkukuk** [comment]
On Tue, Oct 16, Andrew Jeddeloh wrote:

> We can't drop support for the old ones (for compatibility) but we can add
> aliases and support both the historic coreos.* and a new ignition.*. Thoughts?

For compatibility reasons, especially with documentation and common
knowledge, I would go the path with an aliases and deprecating the
coreos.* names.

-- 
Thorsten Kukuk, Distinguished Engineer, Senior Architect SLES & MicroOS
SUSE LINUX GmbH, Maxfeldstr. 5, 90409 Nuernberg, Germany
GF: Felix Imendoerffer, Jane Smithard, Graham Norton, HRB 21284 (AG Nuernberg)

**Comment by lucab** [comment]
I agree with all comments above. I'd say that the final list of IDs to change is:
 * the config URL key
 * the two VMWare keys

Both cases are feasible to just rename as `coreos.config.*` and the old IDs deprecated and kept for compatibility.

**Comment by bgilbert** [comment]
Ignition hasn't been branched for the 3.0 spec, so that PR will break Container Linux.
